linera-core/src/data_types.rs

```diff
@@ -313,7 +313,7 @@ impl<T> ClientOutcome<T> {
         }
     }
 
-    #[expect(clippy::result_large_err)]
+    #[allow(clippy::result_large_err)]
```

Comment:
Do we need `allow` because the lint is triggered with some but not all feature flag combinations?